system/sched_note/note_main.c

```diff
@@ -94,18 +95,17 @@ static void trace_dump_unflatten(FAR void *dst,
     {
 #ifdef CONFIG_HAVE_LONG_LONG
       case 8:
-        *(uint64_t *)dst = ((uint64_t)src[7] << 56)
-                         + ((uint64_t)src[6] << 48)
-                         + ((uint64_t)src[5] << 40)
-                         + ((uint64_t)src[4] << 32);
+        *(uint64_t *)dst = le64toh(*(uint64_t *)src);
```

Comment:
I have only one question. Are we sure that `src` is target type aligned? I mean that if `len == 4` then `src` can be promoted to `uint32_t *`. If not, then we will have troubles with archs that require pointers to be aligned.

Reply:
Yes, there are potential issues if the hardware do not support aligned access, seems there is no more elegant way, I used the scheme you recommended.